src/panels/lovelace/common/evaluate-filter.ts

```diff
@@ -2,11 +2,28 @@ import { HassEntity } from "home-assistant-js-websocket";
 
 export const evaluateFilter = (stateObj: HassEntity, filter: any): boolean => {
   const operator = filter.operator || "==";
-  const value = filter.value ?? filter;
-  const state = filter.attribute
+  let value = filter.value ?? filter;
+  let state = filter.attribute
     ? stateObj.attributes[filter.attribute]
     : stateObj.state;
 
+  if (operator === "==" || operator === "!=") {
+    const valueIsNumeric =
+      typeof value === "number" ||
+      (typeof value === "string" &&
+        !isNaN(+value) &&
+        value.trim().length !== 0);
```

Comment:
```suggestion
    const valueIsNumeric = typeof value === "number" ||
      (value && typeof value === "string" &&
        !isNaN(value));
```

Reply:
I hesitate from this change because it considers strings of whitespace equal to both 0 and "0", and equal to each other.

Your suggestion makes:
```
"" != 0
" " == 0
"   " == 0
" " == "            "
etc.
```

Perhaps I'm being pedantic, but that's why I had the trim().length check. If you still prefer this change even with that caveat, I will make it, but just wanted to point that out, just please reconfirm.

Also `isNan(<string>)` is a lint error, which is why I had the `+` in there originally. If you prefer we can do `isNaN(Number(value))` instead.

Reply:
`isNaN(+value)` is fine here

Reply:
Submitted update.